File: level3.lua
-- Guckets Level 3
-- Made 2007 by Lars Stoltenow <penma@penma.de>
-- License: WTFPL <http://sam.zoy.org/wtfpl>

level = guckets.level:new()
level.author = "Lars Stoltenow"
level.name = "Level 3"

table.insert(level.buckets, guckets.bucket:new({ water = 0, water_max = 3 }))
table.insert(level.buckets, guckets.bucket:new({ water = 0, water_max = 9 }))
table.insert(level.buckets, guckets.bucket:new({ water = 0, water_max = 5 }))

table.insert(level.goals, {
	callback = function (level) return level.buckets[2].water == 7 end,
	text = "%d liters of water in bucket %d",
	arguments = { 7, 2 } })
level:auto_spare_water()

File: data/levels/default/level3.pl
$level = new Guckets::Level();
$level->{author} = "Lars Stoltenow";
$level->{name} = "Level 3";

push(@{$level->{buckets}}, new Guckets::Bucket(0, 4));
push(@{$level->{buckets}}, new Guckets::Bucket(0, 9));

push(@{$level->{goals}}, Guckets::Goals::water(2, 6));

$level->auto_spare_bucket();
